docs/DeveloperGuide.md

```diff
@@ -4,22 +4,194 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Design
+
+#### Architecture
+
+{Description of overall architecture}
+
+#### Ui Component
+
+The structure of the Ui Component is illustrated in the following diagram:
+
+![UiComponentDiagram](images/UiComponent.png)
```

Comment:
I think maybe you could use a class diagram, as it would be more comprehensive in explaining the structure of the Ui component rather than an architechture diagram.

Also, the arrows coming out from bottom component with all the commands look very messy pointing to the same exact point and overlap other components. This may lead to misinterpretations especially if you want to add labels/roles to these arrows. Maybe you could instead use the right angle arrows to go around the bottom component boxes and space them out as along the Command box as well.